src/agentlab/agents/generic_agent/generic_agent.py

```diff
@@ -93,36 +91,31 @@ def get_action(self, obs):
             additional_prompts=system_prompt,
         )
 
-        stats = {}
         try:
             # TODO, we would need to further shrink the prompt if the retry
             # cause it to be too long
 
             chat_messages = [
-                SystemMessage(content=system_prompt),
-                HumanMessage(content=prompt),
+                make_system_message(system_prompt),
+                make_user_message(prompt),
             ]
-            ans_dict = retry_raise(
+            ans_dict = retry(
                 self.chat_llm,
                 chat_messages,
                 n_retry=self.max_retry,
                 parser=main_prompt._parse_answer,
             )
-            # inferring the number of retries, TODO: make this less hacky
-            stats["n_retry"] = (len(chat_messages) - 3) / 2
-            stats["busted_retry"] = 0
         except RetryError as e:
```

Comment:
@recursix i'm not sure how to deal with errors now that ParsingErrors and RetryErrors are raised differently.